Mathlib/Algebra/Category/ContinuousCohomology/Basic.lean

```diff
@@ -0,0 +1,159 @@
+ones. For this to work, we also work in `Action (TopModuleCat R) G`, where the `G` action on `M`
+is only continuous on `M`, and not necessarily continuous in both variables, because the `G` action
+on `C(G, M)` might not be continuous on both variables even if it is on `M`.
+
+## Main definition
+- `ContinuousCohomology.homogeneousCochains`:
+  The functor taking an `R`-linear `G`-representation to the complex of homogeneous cochains.
+- `continuousCohomology`:
+  The functor taking an `R`-linear `G`-representation to its `n`-th continuous homology.
+
+## TODO
+- Show that it coincides with `groupCohomology` for discrete groups.
+- Give the usual description of cochains in terms of `n`-ary functions for locally compact groups.
+- Show that short exact sequences induces long exact sequences in certain scenarios.
+-/
+
+open CategoryTheory ContinuousMap
+
+variable (R G : Type*) [CommRing R] [Group G] [TopologicalSpace R]
+
+namespace ContinuousCohomology
+
+variable [TopologicalSpace G] [IsTopologicalGroup G]
+
+variable {R G} in
+/-- The `G` representation `C(G, rep)` given a representation `rep`.
+The `G` action is defined by `g • f := x ↦ f (g⁻¹ * x)`. -/
+abbrev Iobj (rep : Action (TopModuleCat R) G) : Action (TopModuleCat R) G where
+  V := .of R C(G, rep.V)
+  ρ :=
+  { toFun g := TopModuleCat.ofHom
+      { toFun f := .comp (rep.ρ g).hom (f.comp (Homeomorph.mulLeft g⁻¹))
+        map_add' _ _ := by ext; simp
+        map_smul' _ _ := by ext; simp
+        cont := ((continuous_postcomp _).comp (continuous_precomp _)) }
+    map_one' := ConcreteCategory.ext (by ext; simp)
+    map_mul' _ _ := ConcreteCategory.ext (by ext; simp [mul_assoc]) }
+
+/-- The functor taking a representation `rep` to the representation `C(G, rep)`. -/
+@[simps]
+def I : Action (TopModuleCat R) G ⥤ Action (TopModuleCat R) G where
+  obj := Iobj
+  map {M N} φ :=
+  { hom := TopModuleCat.ofHom (ContinuousLinearMap.compLeftContinuous _ _ φ.hom.hom)
+    comm g := by
+      ext f g'
+      show (M.ρ g ≫ φ.hom).hom (f (g⁻¹ * g')) = (φ.hom ≫ N.ρ g).hom (f (g⁻¹ * g'))
+      rw [φ.comm] }
+  map_id _ := rfl
+  map_comp _ _ := rfl
+
+instance : (I R G).Additive := ⟨by tauto⟩
+instance : (I R G).Linear R := ⟨by tauto⟩
+
+/-- The constant function `rep ⟶ C(G, rep)` as a natural transformation. -/
+@[simps]
+def const : 𝟭 _ ⟶ I R G where
+  app _ := { hom := TopModuleCat.ofHom (.const _ _), comm _ := rfl }
+  naturality _ _ _ := rfl
+
+namespace MultiInd
+
+/-- The n-th functor taking `M` to `C(G, C(G,...,C(G, M)))` (with n `G`s).
+These functors form a complex, see `MultiInd.complex`. -/
+def functor : ℕ → Action (TopModuleCat R) G ⥤ Action (TopModuleCat R) G
+  | 0     => 𝟭 _
+  | n + 1 => functor n ⋙ I R G
+
+/-- The differential map in `MultiInd.complex`. -/
+def d : ∀ n : ℕ, functor R G n ⟶ functor R G (n + 1)
+  | 0     => const R G
+  | n + 1 => whiskerLeft (functor R G (n + 1)) (const R G) - (by exact whiskerRight (d n) (I R G))
```

Comment:
This construction is AFAIK original work of Richard's. I think it deserves at least a comment in the module docstring explaining this insight. I have certainly not read all the continuous group cohomology literature but I'd not seen this before until Richard explained it to me at LLL once.